test/functional/discovered_hosts_controller_test.rb

```diff
@@ -201,48 +202,52 @@ def test_reboot_error
   end
 
   def test_auto_provision_success
+    User.current = User.unscoped.find(set_session_user_default_manager[:user])
```

Comment:
Of course store the old value and put it back, just in case. Alternatively set it in setup and remove in teardown methods, if you like this more.